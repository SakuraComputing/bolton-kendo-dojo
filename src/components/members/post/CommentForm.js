import React, { Component } from 'react'

class commentForm extends Component {
    render() {
        return (
            <div>
                
            </div>
        )
    }
}
export default commentForm;
