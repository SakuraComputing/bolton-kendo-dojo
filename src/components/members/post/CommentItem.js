import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteComment } from '../../../actions/postActions';

export class CommentItem extends Component {

    onDeleteClick(postId, commentId) {
        this.props.deleteComment(postId, commentId);
    }

    render() {
        const { comment, postId, auth } = this.props;

        return(
            <div className="post-container">
                <div className="comment-outer">
                    <div className="comment-inner-1">
                        <a href="profile.html">
                            <img className="rounded-circle d-none d-md-block"
                                    src={comment.avatar} alt=""/>
                        </a>
                        <br/>
                        <p className="avatar-name">{comment.name}</p>
                        <div className="comment-container">
                        </div>
                    </div>
                    <div className="comment-inner-2">
                        <p className="comment-text">{comment.text}</p>
                    </div>
                    <div className="comment-inner-3">
                        {comment.user === auth.user.id ? (
                            <button onClick={this.onDeleteClick.bind(this, postId, comment._id)}
                                    type="button"
                                    className="button-small button-delete item-delete"
                            >
                            <i className="fas fa-times"></i>
                            Delete Comment
                            </button>
                        ) : null}
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    auth: state.auth
});
 
export default connect(mapStateToProps, {deleteComment})(CommentItem);
