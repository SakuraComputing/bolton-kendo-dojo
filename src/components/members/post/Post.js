import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getPost } from '../../../actions/postActions';

export class Post extends Component {
    render() {

        
        return (
            <div>
                
            </div>
        )
    }
}
const mapStateToProps = state => ({
    post: state.post
})
export default connect(mapStateToProps, { getPost })(Post);
