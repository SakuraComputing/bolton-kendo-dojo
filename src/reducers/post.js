import {
    GET_POSTS,
    ADD_POST,
    GET_ERRORS,
    DELETE_POST,
    GET_POST

} from '../actions/types';

const postReducerDefaultState = [];

export default (state = postReducerDefaultState, action) => {
    switch(action.type) {
        
    }
}
