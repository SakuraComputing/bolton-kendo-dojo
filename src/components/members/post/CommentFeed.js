import React, { Component } from 'react';
import CommentItem from './CommentItem';

class commentFeed extends Component {
    render() {
        const {comments, postId} = this.props;

        return comments.map(comment => (
            <CommentItem key={comment._id} comment={comment} postId={postId}/>
        ))
    }
}
export default commentFeed;
