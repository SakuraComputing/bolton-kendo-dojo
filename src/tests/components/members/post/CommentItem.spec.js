import React from 'react';
import { shallow } from 'enzyme';
import { CommentItem } from '../../../../components/members/post/CommentItem';

describe('Comment Item Component', () => {
  let commentItem, comment, auth;

  beforeEach(() => {
      comment = jest.fn();
      auth = { user: jest.fn() };
      commentItem = shallow(<CommentItem comment={comment} auth={auth}/>)
  });
  it('should render correctly', () => {
      expect(commentItem).toMatchSnapshot();
  });
})
