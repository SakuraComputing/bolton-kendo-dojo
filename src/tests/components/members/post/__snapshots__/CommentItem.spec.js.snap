// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Item Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentItem
    auth={
      Object {
        "user": [MockFunction],
      }
    }
    comment={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className=""
      >
        <div
          className=""
        >
          <a
            href="profile.html"
          >
            <img
              alt=""
              className="rounded-circle d-none d-md-block"
            />
          </a>
          <br />
          <p
            className="text-center"
          >
            mockConstructor
          </p>
        </div>
        <div
          className="col-md-10"
        >
          <p
            className="lead"
          />
          <button
            className="button-small button-warning"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-times"
            />
          </button>
        </div>
      </div>,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=""
          >
            <a
              href="profile.html"
            >
              <img
                alt=""
                className="rounded-circle d-none d-md-block"
              />
            </a>
            <br />
            <p
              className="text-center"
            >
              mockConstructor
            </p>
          </div>,
          <div
            className="col-md-10"
          >
            <p
              className="lead"
            />
            <button
              className="button-small button-warning"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-times"
              />
            </button>
          </div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="profile.html"
              >
                <img
                  alt=""
                  className="rounded-circle d-none d-md-block"
                />
              </a>,
              <br />,
              <p
                className="text-center"
              >
                mockConstructor
              </p>,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  className="rounded-circle d-none d-md-block"
                />,
                "href": "profile.html",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "rounded-circle d-none d-md-block",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mockConstructor",
                "className": "text-center",
              },
              "ref": null,
              "rendered": "mockConstructor",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="lead"
              />,
              <button
                className="button-small button-warning"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-times"
                />
              </button>,
            ],
            "className": "col-md-10",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "lead",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-times"
                />,
                "className": "button-small button-warning",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-times",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className=""
        >
          <div
            className=""
          >
            <a
              href="profile.html"
            >
              <img
                alt=""
                className="rounded-circle d-none d-md-block"
              />
            </a>
            <br />
            <p
              className="text-center"
            >
              mockConstructor
            </p>
          </div>
          <div
            className="col-md-10"
          >
            <p
              className="lead"
            />
            <button
              className="button-small button-warning"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-times"
              />
            </button>
          </div>
        </div>,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className=""
            >
              <a
                href="profile.html"
              >
                <img
                  alt=""
                  className="rounded-circle d-none d-md-block"
                />
              </a>
              <br />
              <p
                className="text-center"
              >
                mockConstructor
              </p>
            </div>,
            <div
              className="col-md-10"
            >
              <p
                className="lead"
              />
              <button
                className="button-small button-warning"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fas fa-times"
                />
              </button>
            </div>,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="profile.html"
                >
                  <img
                    alt=""
                    className="rounded-circle d-none d-md-block"
                  />
                </a>,
                <br />,
                <p
                  className="text-center"
                >
                  mockConstructor
                </p>,
              ],
              "className": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="rounded-circle d-none d-md-block"
                  />,
                  "href": "profile.html",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "rounded-circle d-none d-md-block",
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "mockConstructor",
                  "className": "text-center",
                },
                "ref": null,
                "rendered": "mockConstructor",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="lead"
                />,
                <button
                  className="button-small button-warning"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    className="fas fa-times"
                  />
                </button>,
              ],
              "className": "col-md-10",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "lead",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-times"
                  />,
                  "className": "button-small button-warning",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-times",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
